pkg/system/autoscaler.go

```diff
@@ -273,20 +274,23 @@ func (r *Reconciler) autoscaleKeda(prometheus *monitoringv1.Prometheus) error {
 	r.KedaScaled.Spec.Triggers[0].Metadata["serverAddress"] = prometheusURL
 	query := strings.Replace(r.KedaScaled.Spec.Triggers[0].Metadata["query"], "placeholder", r.Request.Namespace, 1)
 	r.KedaScaled.Spec.Triggers[0].Metadata["query"] = query
+	if err := r.ReconcileObject(r.KedaScaled, r.reconcileKedaReplicaCount); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Reconciler) reconcileKedaReplicaCount() error {
 	endpointsSpec := r.NooBaa.Spec.Endpoints
 	var minReplicas int32 = 1
 	var maxReplicas int32 = 2
 	if endpointsSpec != nil {
-		r.KedaScaled.Spec.MinReplicaCount = &endpointsSpec.MinCount
-		r.KedaScaled.Spec.MaxReplicaCount = &endpointsSpec.MaxCount
-	} else {
-		r.KedaScaled.Spec.MinReplicaCount = &minReplicas
-		r.KedaScaled.Spec.MaxReplicaCount = &maxReplicas
-	}
-	if !util.KubeCreateSkipExisting(r.KedaScaled) {
-		log.Errorf("❌ Failed to create KedaScaledObject")
-		return fmt.Errorf("Failed to create KedaScaledObject")
+		minReplicas = endpointsSpec.MinCount
+		maxReplicas = endpointsSpec.MaxCount
 	}
+	fmt.Println("r.KedaScaled.Spec.MinReplicaCount ", minReplicas, maxReplicas)
```

Comment:
Is this printing supposed to be for the user or internal log printing?

Reply:
yes, will remove it